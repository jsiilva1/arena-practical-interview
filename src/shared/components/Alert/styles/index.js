import styled from 'styled-components';

import { device } from '../../../../../theme/settings/screens';

export const AlertContainer = styled.span`
  width: 80%;
  min-width: 10vh;
  background-color: ${(props) => {
    if (props.type === 'error') {
      return '#f8d7da';
    }
    if (props.type === 'alert') {
      return '#FCF2CD';
    }
  }};

  color: ${(props) => {
    if (props.type === 'error') {
      return '#721c24';
    }
  }};

  border: 1px solid;
  border-color: ${(props) => {
    if (props.type === 'error') {
      return '#f5c6cb';
    }
  }};

  padding: 12px 20px;
  border-radius: 0.3rem;
  font-size: 0.9rem;
  font-weight: 600;

  @media ${device.mobile} {
    margin: 0 auto;
  }
`;