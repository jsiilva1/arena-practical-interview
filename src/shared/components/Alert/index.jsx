/**
 * External Dependencies
 */
import React from 'react';

/**
 * Internal Dependencies
 */
import { AlertContainer } from './styles/index';

/**
 * Alert Component
 *
 * @description to render a alert div for messages
 *
 * @param {Object} props props of component
 *
 * @return {*} component.
 */
const Alert = (props) => {
  const {
    type,
    message,
  } = props;

  return (
    <AlertContainer type={type}>
      {message}
    </AlertContainer>
  );
};

export default Alert;
