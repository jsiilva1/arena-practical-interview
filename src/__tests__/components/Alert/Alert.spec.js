import React from 'react';
import { shallow } from 'enzyme';

import Alert from '../../../shared/components/Alert';

describe('Alert Component', () => {
  it('should render the Alert Component correctly', () => {
    const wrapper = shallow(
      <Alert
        message='Message test'
        type="error"
      />
    );
    
    expect(wrapper).toMatchSnapshot();
  });
});
