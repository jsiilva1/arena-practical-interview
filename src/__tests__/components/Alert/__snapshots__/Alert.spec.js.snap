// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Component should render the Alert Component correctly 1`] = `
<styled.span
  type="error"
>
  Message test
</styled.span>
`;
